grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -263,8 +265,7 @@ class StockMovementService {
     void updateRequisitionStatus(String id, RequisitionStatus status, Comment comment = null) {
 
         log.info "Update status ${id} " + status
-        // TODO: In Grails the get below should be replaced by the data service get that joins the Events
-        Requisition requisition = Requisition.get(id)
+        Requisition requisition = requisitionDataService.getRequisitionWithEvents(id)
```

Comment:
ok, since we joined the events here, shouldn't we remove the additional fetch from somewhere, that was added in order to prevent the LIE?

Reply:
@drodzewicz hm, I guess nevermind, but it's weird that here, I see the `Requisition.get`, but when I go to your commit directly, I can see this:
![Screenshot from 2023-12-11 18-07-16](https://github.com/openboxes/openboxes/assets/93163821/749aea0c-123b-4dd3-b06b-c2a1c990e44f)

which would answer what I wrote, but got mistaken by the change I can actually see in "Changed files"

Reply:
there was not additional fetch before the hotfix, everything is correct.
The additional fetch was in the first PR for this hotfix which added the `Requisition.get(event.requsistion.id)` in `onApplicationEvent(RequisitionStatusTransitionEvent event)`